src/file/File.cpp

```diff
@@ -40,7 +40,16 @@ File::File(FileId id, std::string fileName, int flags, int fd, shared_ptr<Active
 }
 
 int64_t File::read(char *buffer, int64_t length) {
-    return mInStream->read(buffer, length);
+    int64_t bytesRead = length < remainingLength() ? length : remainingLength();
+    if(bytesRead < 0){
```

Comment:
the input parm length should already guaranteed in Gopherwood.cpp, so it's greater than 0.
remaining should also guaranteed inside function remaining()
so here we should never get a bytesToRead smaller than 0